src/dataflow-types/src/types.rs

```diff
@@ -366,6 +366,39 @@ where
     }
 }
 
+impl<P, T> DataflowDescription<P, T> {
+    /// Determine a unique id for this dataflow based on the items it produces.
+    pub fn global_id(&self) -> Option<GlobalId> {
```

Comment:
I can't exactly tell what is going on here. I was imagining using the ids of the things we export (e.g. indexes or sinks) as the ids, rather than the ids of intermediate views. We could, for example, have two dataflows that both rely on the same intermediate view, with some low id, but build two different indexes off of it.

I don't think this will be a unique id, as written.

Reply:
Yep, I first had a different variant that included the index exports and descriptions, and "fixed" it by removing the correct thing. Should be better now, but don't feel like testing on the laptop right now :)